### PR TITLE
Set proper network_type for the floating network

### PR DESCRIPTION
**Pull request by dirkmueller**
The floating network is never a GRE type network, its either
VLAN or flat.

### CODE REVIEW

**Comment by rossella** [comment]
LGTM

File: chef/cookbooks/neutron/recipes/post_install_conf.rb [hunk 1]
```diff
@@ -84,7 +84,11 @@ def mask_to_bits(mask)
     end
   when 'gre'
     fixed_network_type = "--provider:network_type gre --provider:segmentation_id 1"
-    floating_network_type = "--provider:network_type gre --provider:segmentation_id 2"
+    if node[:network][:networks][:nova_floating][:use_vlan]
```
Comment (rossella) [inline chef/cookbooks/neutron/recipes/post_install_conf.rb]: maybe this should be moved after line 78? We should take care of the floating network just once not for every type (gre, vxlan, etc) so we can remove some lines like 81-83 etc

Comment (dirkmueller) [inline chef/cookbooks/neutron/recipes/post_install_conf.rb]: yeah, I was wondering about that- I'm not sure about the vxlan case (see below).

Comment (toabctl) [inline chef/cookbooks/neutron/recipes/post_install_conf.rb]: > yeah, I was wondering about that- I'm not sure about the vxlan case (see below).

For provider networks, I think only `flat` and `vlan` are useful. From http://docs.openstack.org/admin-guide-cloud/content/tenant-provider-networks.html : 
```
Provider networks. Provider networks are created by the OpenStack administrator and map directly to an existing physical network in the data center. Useful network types in this category are flat (untagged) and VLAN (802.1Q tagged). It is possible to allow provider networks to be shared among tenants as part of the network creation process.
```

Comment (dirkmueller) [inline chef/cookbooks/neutron/recipes/post_install_conf.rb]: we can try the other variant as well..

**Comment by rossella** [comment]
LGTM now

**Comment by rsalevsky** [comment]
LGTM

**Pull request merged**